### PR TITLE
Allow Firebase to be configured for cross-platform apps

### PR DESCRIPTION
**Pull request by kitallis**
Configure Android and iOS apps on FAD separately. Remove the need for App ID config while configuring the integration.

Screenshot 2023-06-30 at 3 24 57 AM

### CODE REVIEW

File: db/migrate/20230628115739_add_platform_specific_integration_config.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class AddPlatformSpecificIntegrationConfig < ActiveRecord::Migration[7.0]
+  def change
+    safety_assured do
+      change_table :deployments, bulk: true do |t|
```
Comment (nid90) [inline db/migrate/20230628115739_add_platform_specific_integration_config.rb:4]: I think this is a typo. the table being changed is not `deployments`.

File: app/models/app_config.rb [hunk 2]
```diff
@@ -2,44 +2,77 @@
...
+  validates :firebase_ios_config,
+    allow_blank: true,
+    json: {message: ->(errors) { errors }, schema: FIREBASE_CONFIG_SCHEMA}
+  validates :firebase_android_config,
+    allow_blank: true,
+    json: {message: ->(errors) { errors }, schema: FIREBASE_CONFIG_SCHEMA}
+
   def ready?
-    MINIMUM_REQUIRED_CONFIG.all? { |config| public_send(config).present? }
+    MINIMUM_REQUIRED_CONFIG.all? { |config| public_send(config).present? } && firebase_ready?
```
Comment (nid90) [inline app/models/app_config.rb:31]: does firebase always have to be ready?

File: app/models/app_config.rb [hunk 3]
```diff
@@ -2,44 +2,77 @@
...
+    elsif app.ios?
+      {ios: ios_apps}
+    elsif app.cross_platform?
+      {ios: ios_apps, android: android_apps}
+    end
+  end
+
+  private
+
+  def firebase_ready?
```
Comment (kitallis) [inline app/models/app_config.rb:73]:
```suggestion
  def firebase_ready?
    return true if app.integrations.google_firebase_integrations.none?
```

File: app/models/app_config.rb (hunk 2 above)
Comment (kitallis) [reply app/models/app_config.rb:31]: Fixed.

**Review by nid90 — APPROVED** [review]

**Pull request merged**